### ISSUE TITLE
tidyverse makes locking fail because of data.table dependency

### ISSUE THREAD

**Issue by notPlancha** [AstraZeneca/roo#82]
My best guess is because data.table is not a package found on CRAN
```
PS C:\Users\<user>\project> roo lock
Lockfile not found.
  - tidyverse (2.0.0)
    - broom (1.0.4)
      - backports (1.4.1)
      - dplyr (1.1.2)
        - cli (3.6.1)
        - generics (0.1.3)
        - glue (1.6.2)
        - lifecycle (1.0.3)
          - cli (...)
          - glue (...)
          - rlang (1.1.1)
        - magrittr (2.0.3)
        - pillar (1.9.0)
          - cli (...)
          - fansi (1.0.4)
          - glue (...)
          - lifecycle (...)
          - rlang (...)
          - utf8 (1.2.3)
          - vctrs (0.6.2)
            - cli (...)
            - glue (...)
            - lifecycle (...)
            - rlang (...)
        - R6 (2.5.1)
        - rlang (...)
        - tibble (3.2.1)
          - fansi (...)
          - lifecycle (...)
          - magrittr (...)
          - pillar (...)
          - pkgconfig (2.0.3)
          - rlang (...)
          - vctrs (...)
        - tidyselect (1.2.0)
          - cli (...)
          - glue (...)
          - lifecycle (...)
          - rlang (...)
          - vctrs (...)
          - withr (2.5.0)
        - vctrs (...)
      - ellipsis (0.3.2)
        - rlang (...)
      - generics (...)
      - glue (...)
      - lifecycle (...)
      - purrr (1.0.1)
        - cli (...)
        - lifecycle (...)
        - magrittr (...)
        - rlang (...)
        - vctrs (...)
      - rlang (...)
      - stringr (1.5.0)
        - cli (...)
        - glue (...)
        - lifecycle (...)
        - magrittr (...)
        - rlang (...)
        - stringi (1.7.12)
        - vctrs (...)
      - tibble (...)
      - tidyr (1.3.0)
        - cli (...)
        - dplyr (...)
        - glue (...)
        - lifecycle (...)
        - magrittr (...)
        - purrr (...)
        - rlang (...)
        - stringr (...)
        - tibble (...)
        - tidyselect (...)
        - vctrs (...)
        - cpp11 (0.4.3)
    - conflicted (1.2.0)
      - cli (...)
      - memoise (2.0.1)
        - rlang (...)
        - cachem (1.0.8)
          - rlang (...)
          - fastmap (1.1.1)
      - rlang (...)
    - cli (...)
    - dbplyr (2.3.2)
      - blob (1.2.4)
        - rlang (...)
        - vctrs (...)
      - cli (...)
      - DBI (1.1.3)
      - dplyr (...)
      - glue (...)
      - lifecycle (...)
      - magrittr (...)
      - pillar (...)
      - purrr (...)
      - R6 (...)
      - rlang (...)
      - tibble (...)
      - tidyr (...)
      - tidyselect (...)
      - vctrs (...)
      - withr (...)
    - dplyr (...)
    - dtplyr (1.3.1)
      - cli (...)
Error: Unable to create lock file: Unable to find package for dependency data.table
PS C:\Users\<user>\project> cat .\rproject.toml
[tool.roo]
name = "mans"
version = "1.0.0"
[[tool.roo.source]]
name = "CRAN"
url = "https://cran.radicaldevelop.com/"
[tool.roo.dependencies]
tidyverse = "*"
```

**Comment by stefanoborini** [comment]
That's interesting. It should be there. I'll check and come back to you.

**Comment by stefanoborini** [comment]
Ok, I traced it down. I'll fix it today and release 0.17.1. I'll keep you posted.

**Comment by stefanoborini** [comment]
It's fixed now. The reason it didn't work is that I exceeded in caution. 

Roo needs to get the list of available packages from the src/contrib content. This is a webpage that I need to scrape. To find the packages I extract the a href name, but also check if it matches with the content of the link. 

I assumed that these always matched, because that's what happens on https://cran.r-project.org/src/contrib/. But in your mirror, if a filename is too long, it's cut with an ellipsis ...

So for some packages, the name was not matching and I rejected the package as something that was not a package.

I removed the match check and now it works. It's version 0.17.1 and it's already on pypi.

Thanks!
